neptune/new/internal/utils/deprecation.py

```diff
@@ -19,7 +19,17 @@
 
 from neptune.new.exceptions import NeptuneParametersCollision
 
-__all__ = ["deprecated", "deprecated_parameter"]
+__all__ = ["deprecated", "deprecated_parameter", "simple_warning"]
+
+
+def simple_warning(message: str, stack_level: int):
+    warnings.simplefilter("once", DeprecationWarning)
+    warnings.warn(
+        message,
+        category=DeprecationWarning,
+        stacklevel=stack_level,
+    )
+    warnings.simplefilter("default", DeprecationWarning)
```

Comment:
Stan w tej libce jest per wątek?

Reply:
There was a problems with `simplefilter` and `GitPython` (probably due to running `git` in a subprocesses). I've rewritten it to simple dictionary storage on the function level (internally it's almost the same as `warnings` library works). Operations on dict in Python should be probably atomic and thread-safe but there is logic like `if not exists then add` so probably several warnings could be called in some race conditions but <=1 per-thread (I think so). It still almost the same as `warnings.simplefilter("once", ...)`.